| 単位名称         | 記号 | バイト換算           | 累乗表記     |
| :--------------- | :--- | :------------------- | :----------- |
| Kilobyte         | kB   | 1,000 B              | 10<sup>3</sup> B  |
| Megabyte         | MB   | 1,000,000 B          | 10<sup>6</sup> B  |
| Gigabyte         | GB   | 1,000,000,000 B      | 10<sup>9</sup> B  |
| Terabyte         | TB   | 1,000,000,000,000 B  | 10<sup>12</sup> B |

| 単位名称         | 記号 | バイト換算             | 累乗表記     |
| :--------------- | :--- | :--------------------- | :----------- |
| Kibibyte         | KiB  | 1,024 B                | 2<sup>10</sup> B |
| Mebibyte         | MiB  | 1,048,576 B            | 2<sup>20</sup> B |
| Gibibyte         | GiB  | 1,073,741,824 B        | 2<sup>30</sup> B |
| Tebibyte         | TiB  | 1,099,511,627,776 B    | 2<sup>40</sup> B |

| 日本語   | 英語    | 記号 | 10進数表記         | 累乗表記        |
| :------- | :------ | :--- | :----------------- | :-------------- |
| テラ     | Tera    | T    | 1,000,000,000,000 | 10<sup>12</sup> |
| ギガ     | Giga    | G    | 1,000,000,000     | 10<sup>9</sup>  |
| メガ     | Mega    | M    | 1,000,000         | 10<sup>6</sup>  |
| キロ     | Kilo    | k    | 1,000             | 10<sup>3</sup>  |
| (基準)   | (Base)  | -    | 1                 | 10<sup>0</sup>  |
| ミリ     | Milli   | m    | 0.001             | 10<sup>-3</sup> |
| マイクロ | Micro   | μ    | 0.000001          | 10<sup>-6</sup> |
| ナノ     | Nano    | n    | 0.000000001       | 10<sup>-9</sup> |
| ピコ     | Pico    | p    | 0.000000000001    | 10<sup>-12</sup> |

| 進数 | 基数 | 使用する数字 | 用途 |
| :--- | :--- | :--- | :--- |
| **10進数** (Decimal) | 10 | 0,1,2,3,4,5,6,7,8,9 | 日常生活、一般的な計算 |
| **2進数** (Binary) | 2 | 0,1 | コンピュータの内部処理（全てのデータの根本） |
| **16進数** (Hexadecimal) | 16 | 0-9, A(10),B(11),C(12),D(13),E(14),F(15) | 2進数を簡潔に表現（メモリアドレス、色コード等） |

In [5]:
def binary_to_decimal_detailed(bin_str):
    """
    将二进制字符串转换为十进制数，并详细输出每一步的计算过程。
    """
    # 检查输入是否有效
    if not all(c in '01.' for c in bin_str):
        print("错误：输入必须只包含0、1和小数点（.）")
        return
    
    # 分离整数部分和小数部分
    if '.' in bin_str:
        integer_part, fractional_part = bin_str.split('.')
    else:
        integer_part = bin_str
        fractional_part = ''
    
    print(f"输入二进制字符串: {bin_str}")
    print("转换过程：")
    
    total = 0.0
    steps = []  # 存储每一步的详细描述和值
    
    # 处理整数部分
    if integer_part:
        print("\n整数部分计算:")
        int_length = len(integer_part)
        for i, digit in enumerate(integer_part):
            power = int_length - i - 1  # 指数从高位到低位递减
            value = int(digit) * (2 ** power)
            total += value
            step_desc = f"{digit} × 2^{power} = {value}"
            steps.append(step_desc)
            print(step_desc)
    else:
        print("\n整数部分为空，跳过计算。")
    
    # 处理小数部分
    if fractional_part:
        print("\n小数部分计算:")
        for i, digit in enumerate(fractional_part):
            power = -(i + 1)  # 指数从-1开始递减
            value = int(digit) * (2 ** power)
            total += value
            step_desc = f"{digit} × 2^{power} = {value}"
            steps.append(step_desc)
            print(step_desc)
    else:
        print("\n小数部分为空，跳过计算。")
    
    # 输出总和公式
    print("\n总和公式:")
    formula = " + ".join([step.split(' = ')[1] for step in steps])
    print(formula + f" = {total}")
    
    print(f"\n最终结果: {bin_str}₂ = {total}₁₀")
    return total

# 示例使用
if __name__ == "__main__":
    # 测试几个例子
    examples = ["101001101.101"]
    for example in examples:
        print("="*50)
        binary_to_decimal_detailed(example)
        print("="*50)

输入二进制字符串: 101001101.101
转换过程：

整数部分计算:
1 × 2^8 = 256
0 × 2^7 = 0
1 × 2^6 = 64
0 × 2^5 = 0
0 × 2^4 = 0
1 × 2^3 = 8
1 × 2^2 = 4
0 × 2^1 = 0
1 × 2^0 = 1

小数部分计算:
1 × 2^-1 = 0.5
0 × 2^-2 = 0.0
1 × 2^-3 = 0.125

总和公式:
256 + 0 + 64 + 0 + 0 + 8 + 4 + 0 + 1 + 0.5 + 0.0 + 0.125 = 333.625

最终结果: 101001101.101₂ = 333.625₁₀


In [8]:
# 【実践計算】10進数 → 16進数 変換（詳細版）
def decimal_to_hex_detailed(decimal_num, precision=6):
    """
    10進数を16進数に変換し、その過程を詳細に表示する関数。
    """
    # 整数部と小数部を分離
    integer_part = int(decimal_num)
    fractional_part = decimal_num - integer_part

    print(f"変換元: {decimal_num}")
    hex_digits = "0123456789ABCDEF" # 16進数の桁リスト
    result_int = [] # 整数部の結果格納用リスト
    result_frac = [] # 小数部の結果格納用リスト

    # --- 整数部の変換プロセス ---
    print("\n=== 整数部の計算 ===")
    n = integer_part
    if n == 0:
        result_int.append('0')
    else:
        step_count = 0
        while n > 0:
            step_count += 1
            remainder = n % 16
            n //= 16
            hex_digit = hex_digits[remainder]
            result_int.insert(0, hex_digit) # 結果をリストの先頭に追加
            print(f"ステップ{step_count}: {n} * 16 + {remainder} [→ {hex_digit}]")

    # --- 小数部の変換プロセス ---
    print("\n=== 小数部の計算 ===")
    f = fractional_part
    step_count = 0
    while f > 0 and step_count < precision:
        step_count += 1
        f *= 16
        integer_component = int(f)
        f -= integer_component
        hex_digit = hex_digits[integer_component]
        result_frac.append(hex_digit)
        print(f"ステップ{step_count}: {fractional_part:.6f} * 16 = {f:.6f} → 整数部分: {integer_component} [→ {hex_digit}], 新しい小数部分: {f:.6f}")

    # 結果の結合と表示
    hex_str = ''.join(result_int)
    if result_frac:
        hex_str += '.' + ''.join(result_frac)
    print(f"\n変換結果: {decimal_num}₁₀ = {hex_str}₁₆")
    return hex_str

# 例の実行
decimal_to_hex_detailed(10283)

変換元: 10283

=== 整数部の計算 ===
ステップ1: 642 * 16 + 11 [→ B]
ステップ2: 40 * 16 + 2 [→ 2]
ステップ3: 2 * 16 + 8 [→ 8]
ステップ4: 0 * 16 + 2 [→ 2]

=== 小数部の計算 ===

変換結果: 10283₁₀ = 282B₁₆


'282B'

| 2進数 (4ビット) | 10進数 | 16進数 |
| :-------------- | :----- | :----- |
| 0000            | 0      | 0      |
| 0001            | 1      | 1      |
| 0010            | 2      | 2      |
| 0011            | 3      | 3      |
| 0100            | 4      | 4      |
| 0101            | 5      | 5      |
| 0110            | 6      | 6      |
| 0111            | 7      | 7      |
| 1000            | 8      | 8      |
| 1001            | 9      | 9      |
| 1010            | 10     | A      |
| 1011            | 11     | B      |
| 1100            | 12     | C      |
| 1101            | 13     | D      |
| 1110            | 14     | E      |
| 1111            | 15     | F      |

In [9]:
# 【実践計算】2進数 → 16進数 変換（グループ化の可視化）
def binary_to_hex_visual(bin_str):
    """
    2進数文字列を16進数に変換し、グループ化の過程を詳細に表示する。
    """
    # 対応表の定義
    bin_to_hex_dict = {
        '0000': '0', '0001': '1', '0010': '2', '0011': '3',
        '0100': '4', '0101': '5', '0110': '6', '0111': '7',
        '1000': '8', '1001': '9', '1010': 'A', '1011': 'B',
        '1100': 'C', '1101': 'D', '1110': 'E', '1111': 'F'
    }

    # 整数部と小数部に分割
    if '.' in bin_str:
        integer_part, fractional_part = bin_str.split('.')
    else:
        integer_part, fractional_part = bin_str, ''

    print(f"入力: {bin_str}")
    hex_result = []

    # --- 整数部の処理 ---
    print("\n1. 整数部のグループ化と変換:")
    # 整数部を4桁ごとのグループに分割（左端はゼロ埋め）
    int_length = len(integer_part)
    # 4の倍数になるまで左側をゼロ埋め
    padded_int = integer_part.zfill((int_length + 3) // 4 * 4)
    # 4桁ずつに分割
    int_groups = [padded_int[i:i+4] for i in range(0, len(padded_int), 4)]
    print(f"   グループ: {int_groups}")

    for group in int_groups:
        hex_digit = bin_to_hex_dict[group]
        hex_result.append(hex_digit)
        print(f"   {group} -> {hex_digit}")

    # --- 小数部の処理 ---
    if fractional_part:
        hex_result.append('.') # 小数点を追加
        print("\n2. 小数部のグループ化と変換:")
        # 小数部を4桁ごとのグループに分割（右端はゼロ埋め）
        frac_length = len(fractional_part)
        padded_frac = fractional_part.ljust((frac_length + 3) // 4 * 4, '0')
        frac_groups = [padded_frac[i:i+4] for i in range(0, len(padded_frac), 4)]
        print(f"   グループ: {frac_groups}")

        for group in frac_groups:
            hex_digit = bin_to_hex_dict[group]
            hex_result.append(hex_digit)
            print(f"   {group} -> {hex_digit}")

    # 結果を結合して表示
    final_hex = ''.join(hex_result)
    # 小数部の末尾の不要な'0'を削除（オプション）
    if '.' in final_hex:
        final_hex = final_hex.rstrip('0').rstrip('.')
    print(f"\n最終結果: {bin_str}₂ = {final_hex}₁₆")
    return final_hex

# 例の実行
binary_to_hex_visual("1101011100.10111")

入力: 1101011100.10111

1. 整数部のグループ化と変換:
   グループ: ['0011', '0101', '1100']
   0011 -> 3
   0101 -> 5
   1100 -> C

2. 小数部のグループ化と変換:
   グループ: ['1011', '1000']
   1011 -> B
   1000 -> 8

最終結果: 1101011100.10111₂ = 35C.B8₁₆


'35C.B8'

| n   | 16^n (整数)       | 16^-n (小数)          |
|:----|------------------:|----------------------:|
| 0   | 1                 | 1                     |
| 1   | 16                | 0.0625                |
| 2   | 256               | 0.00390625            |
| 3   | 4096              | 0.000244140625        |
| 4   | 65536             | 0.0000152587890625    |

In [13]:
# 【実践計算】16進数 → 10進数 変換（詳細版）
def hex_to_decimal_detailed(hex_str):
    """
    16進数字符列を10進数に変換し、その過程を詳細に表示する関数。
    """
    # 16進数から10進数への文字変換辞書
    hex_map = {'0':0, '1':1, '2':2, '3':3, '4':4, '5':5, '6':6, '7':7,
               '8':8, '9':9, 'A':10, 'B':11, 'C':12, 'D':13, 'E':14, 'F':15,
               'a':10, 'b':11, 'c':12, 'd':13, 'e':14, 'f':15}

    # 整数部と小数部に分割
    if '.' in hex_str:
        integer_part, fractional_part = hex_str.split('.')
    else:
        integer_part = hex_str
        fractional_part = ''

    print(f"変換元: {hex_str}")
    total = 0.0
    steps = []  # 計算ステップを記録

    # --- 整数部の処理 ---
    print("\n=== 整数部の計算 ===")
    int_length = len(integer_part)
    for i, char in enumerate(integer_part):
        digit_value = hex_map[char] # 文字を数値に変換 (e.g., 'A' -> 10)
        power = int_length - i - 1   # 指数を計算 (左から)
        weight = 16 ** power         # 重みを計算 (16^power)
        product = digit_value * weight # 数字と重みの積
        total += product

        # ステップの詳細を記録と表示
        step_desc = f"'{char}'({digit_value}) × 16^{power} ({weight}) = {product}"
        steps.append(step_desc)
        print(step_desc)

    # --- 小数部の処理 ---
    if fractional_part:
        print("\n=== 小数部の計算 ===")
        for i, char in enumerate(fractional_part, start=1):
            digit_value = hex_map[char]
            power = -i               # 指数を計算 (-1, -2, ...)
            weight = 16 ** power     # 重みを計算 (16^-power)
            product = digit_value * weight
            total += product

            # ステップの詳細を記録と表示
            step_desc = f"'{char}'({digit_value}) × 16^{power} ({weight:.8f}) = {product:.8f}"
            steps.append(step_desc)
            print(step_desc)

    # 結果を表示
    print(f"\n総和: {total}")
    print(f"変換結果: {hex_str}₁₆ = {total}₁₀")
    return total

# 例の実行
hex_to_decimal_detailed("7C2A.F84")

変換元: 7C2A.F84

=== 整数部の計算 ===
'7'(7) × 16^3 (4096) = 28672
'C'(12) × 16^2 (256) = 3072
'2'(2) × 16^1 (16) = 32
'A'(10) × 16^0 (1) = 10

=== 小数部の計算 ===
'F'(15) × 16^-1 (0.06250000) = 0.93750000
'8'(8) × 16^-2 (0.00390625) = 0.03125000
'4'(4) × 16^-3 (0.00024414) = 0.00097656

総和: 31786.9697265625
変換結果: 7C2A.F84₁₆ = 31786.9697265625₁₀


31786.9697265625

In [14]:
def decimal_to_binary_detailed(decimal_num, precision=8):
    """
    10進数を2進数に変換し、詳細なプロセスを表示する
    """
    # 整数部と小数部を分離
    integer_part = int(decimal_num)
    fractional_part = decimal_num - integer_part
    
    print(f"変換元: {decimal_num}")
    binary_int = []  # 整数部の2進数
    binary_frac = [] # 小数部の2進数
    
    # === 整数部の変換プロセス ===
    print("\n=== 整数部の変換 ===")
    n = integer_part
    step = 1
    while n > 0:
        quotient = n // 2
        remainder = n % 2
        binary_int.insert(0, str(remainder))  # 先頭に追加（逆順保存）
        
        print(f"ステップ{step}: {n} ÷ 2 = {quotient} ... 余り {remainder}")
        n = quotient
        step += 1
    
    # 整数部が0の場合の処理
    if not binary_int:
        binary_int = ['0']
    
    # === 小数部の変換プロセス ===
    print("\n=== 小数部の変換 ===")
    f = fractional_part
    step = 1
    while f > 0 and len(binary_frac) < precision:
        product = f * 2
        integer_part_product = int(product)
        f = product - integer_part_product
        
        binary_frac.append(str(integer_part_product))
        print(f"ステップ{step}: {f:.8f} × 2 = {product:.8f} → 整数部分 {integer_part_product}")
        step += 1
    
    # 結果の組み立て
    binary_str = ''.join(binary_int)
    if binary_frac:
        binary_str += '.' + ''.join(binary_frac)
    
    print(f"\n変換結果: {decimal_num}₁₀ = {binary_str}₂")
    return binary_str

# 実行例
decimal_to_binary_detailed(13.625)

変換元: 13.625

=== 整数部の変換 ===
ステップ1: 13 ÷ 2 = 6 ... 余り 1
ステップ2: 6 ÷ 2 = 3 ... 余り 0
ステップ3: 3 ÷ 2 = 1 ... 余り 1
ステップ4: 1 ÷ 2 = 0 ... 余り 1

=== 小数部の変換 ===
ステップ1: 0.25000000 × 2 = 1.25000000 → 整数部分 1
ステップ2: 0.50000000 × 2 = 0.50000000 → 整数部分 0
ステップ3: 0.00000000 × 2 = 1.00000000 → 整数部分 1

変換結果: 13.625₁₀ = 1101.101₂


'1101.101'

In [15]:
# 【実践計算】16進数 → 2進数 変換（詳細版）
def hex_to_binary_detailed(hex_str):
    """
    16進数を2進数に変換し、詳細なプロセスを表示する関数
    """
    # 対応表
    hex_to_bin_dict = {
        '0': '0000', '1': '0001', '2': '0010', '3': '0011',
        '4': '0100', '5': '0101', '6': '0110', '7': '0111',
        '8': '1000', '9': '1001', 'A': '1010', 'B': '1011',
        'C': '1100', 'D': '1101', 'E': '1110', 'F': '1111',
        'a': '1010', 'b': '1011', 'c': '1100', 'd': '1101',
        'e': '1110', 'f': '1111'
    }
    
    print(f"変換元: {hex_str}₁₆")
    
    # 整数部と小数部に分割
    if '.' in hex_str:
        integer_part, fractional_part = hex_str.split('.')
    else:
        integer_part, fractional_part = hex_str, ''
    
    binary_result = []
    steps = []
    
    # --- 整数部の変換 ---
    print("\n=== 整数部の変換 ===")
    for i, char in enumerate(integer_part):
        bin_digits = hex_to_bin_dict[char]
        binary_result.append(bin_digits)
        steps.append(f"'{char}' → {bin_digits}")
        print(f"  {char} → {bin_digits}")
    
    # 小数部がある場合は小数点を追加
    if fractional_part:
        binary_result.append('.')
        print("\n=== 小数部の変換 ===")
        for i, char in enumerate(fractional_part):
            bin_digits = hex_to_bin_dict[char]
            binary_result.append(bin_digits)
            steps.append(f"'{char}' → {bin_digits}")
            print(f"  {char} → {bin_digits}")
    
    # 結果を結合
    raw_binary = ''.join(binary_result)
    
    # 不要な0を除去
    # 整数部の先頭の0を除去
    if '.' in raw_binary:
        int_part, frac_part = raw_binary.split('.')
        int_part = int_part.lstrip('0') or '0'  # 全て0の場合は'0'を残す
        # 小数部の末尾の0を除去
        frac_part = frac_part.rstrip('0')
        final_binary = f"{int_part}.{frac_part}" if frac_part else int_part
    else:
        final_binary = raw_binary.lstrip('0') or '0'
    
    # 変換過程と結果を表示
    print("\n=== 変換過程 ===")
    print(" -> ".join(steps))
    print(f"\n生の2進数: {raw_binary}")
    print(f"最適化後: {final_binary}")
    print(f"\n最終結果: {hex_str}₁₆ = {final_binary}₂")
    return final_binary

# 例の実行
hex_to_binary_detailed("3A.7C")

変換元: 3A.7C₁₆

=== 整数部の変換 ===
  3 → 0011
  A → 1010

=== 小数部の変換 ===
  7 → 0111
  C → 1100

=== 変換過程 ===
'3' → 0011 -> 'A' → 1010 -> '7' → 0111 -> 'C' → 1100

生の2進数: 00111010.01111100
最適化後: 111010.011111

最終結果: 3A.7C₁₆ = 111010.011111₂


'111010.011111'

| 表現方法       | 日本語         | 中国語   | 核心原理                                                                 |
|:---------------|:---------------|:---------|:-------------------------------------------------------------------------|
| 絶対値表現     | 絶対値表現     | 原码     | 先頭1ビットを符号（0:正, 1:負）、残りを絶対値で表現                         |
| 1の補数表現    | 1の補数表現    | 反码     | 正数のビットを全て反転（0↔1）して負数を表現                                |
| 2の補数表現    | 2の補数表現    | 补码     | 正数のビットを全て反転し、さらに1を加算して負数を表現                       |

| 操作     | ルール                     | 具体例 (8ビット)          |
|:---------|:---------------------------|:--------------------------|
| 左シフト | はみ出たビットは捨て、右端に0を埋める | 00101100 << 2 → 10110000  |
| 右シフト | はみ出たビットは捨て、左端に0を埋める | 00101100 >> 2 → 00001011  |

| 操作     | ルール                                       | 具体例 (8ビット)          |
|:---------|:---------------------------------------------|:--------------------------|
| 左シフト | はみ出たビットは捨て、右端に0を埋める<br>(論理シフトと同じ) | 10101100 << 2 → 10110000  |
| 右シフト | はみ出たビットは捨て、左端に符号ビットと同じ値を埋める     | 10101100 >> 2 → 11101011  |

| 記号 | 名称           | 意味                     |
|:-----|:---------------|:-------------------------|
| ∩    | かつ (AND)     | 積集合 (共通部分)        |
| ∪    | または (OR)    | 和集合                   |
| ⊂    | ではない (NOT) | 補集合                   |
| ∈    | 属する         | 要素が集合に含まれる     |

| A | B | A AND B |
|:-:|:-:|:-------:|
| 0 | 0 |    0    |
| 0 | 1 |    0    |
| 1 | 0 |    0    |
| 1 | 1 |    1    |

| A | B | A OR B |
|:-:|:-:|:------:|
| 0 | 0 |   0    |
| 0 | 1 |   1    |
| 1 | 0 |   1    |
| 1 | 1 |   1    |

XOR演算 (排他的論理和)​ - AとBが異なる場合に「1」、同じ場合に「0」になる
記号: ⊕

| A | B | A XOR B |
|:-:|:-:|:-------:|
| 0 | 0 |    0    |
| 0 | 1 |    1    |
| 1 | 0 |    1    |
| 1 | 1 |    0    |

| A | NOT A |
|:-:|:-----:|
| 0 |   1   |
| 1 |   0   |

| A | B | A AND B | A NAND B |
|:-:|:-:|:-------:|:--------:|
| 0 | 0 |    0    |     1    |
| 0 | 1 |    0    |     1    |
| 1 | 0 |    0    |     1    |
| 1 | 1 |    1    |     0    |

NOR演算 (否定論理和 / NOT OR)​
計算方法​​: OR演算の結果を​​反転​​させる
号​​: ⊽ または ↓

| A | B | A OR B | A NOR B |
|:-:|:-:|:------:|:-------:|
| 0 | 0 |   0    |    1    |
| 0 | 1 |   1    |    0    |
| 1 | 0 |   1    |    0    |
| 1 | 1 |   1    |    0    |

| 優先順位 | 演算子      | 名称   | 説明                             |
|:---------|:------------|:-------|:---------------------------------|
| 最高     | ( )         | 括弧   | 括弧内の式を最優先で評価           |
| 高       | NOT (¬)     | 否定   | 単項演算子。対象を反転             |
| 中       | AND (・)    | 論理積 | 両方の条件が真の場合に真           |
| 低       | OR (+)      | 論理和 | どちらかの条件が真の場合に真       |

| A | B | SUM | CARRY |
|:-:|:-:|:---:|:-----:|
| 0 | 0 |  0  |   0   |
| 0 | 1 |  1  |   0   |
| 1 | 0 |  1  |   0   |
| 1 | 1 |  0  |   1   |

| A | B | C_in | SUM | C_out |
|:-:|:-:|:----:|:---:|:-----:|
| 0 | 0 |  0   |  0  |   0   |
| 0 | 0 |  1   |  1  |   0   |
| 0 | 1 |  0   |  1  |   0   |
| 0 | 1 |  1   |  0  |   1   |
| 1 | 0 |  0   |  1  |   0   |
| 1 | 0 |  1   |  0  |   1   |
| 1 | 1 |  0   |  0  |   1   |
| 1 | 1 |  1   |  1  |   1   |

| 表記法   | 日本語                 | 英語                      | 特徴                                             |
|:---------|:-----------------------|:--------------------------|:-------------------------------------------------|
| 中置記法 | 中置記法<br>(ちゅうちきほう) | Infix Notation            | 演算子がオペランドの間に位置する<br>人間が日常的に使用する表記法 |
| 前置記法 | 前置記法<br>(ぜんちきほう) | Prefix Notation<br>(Polish Notation) | 演算子がオペランドの前に位置する<br>括弧が不要になる利点がある |
| 後置記法 | 後置記法<br>(こうちきほう) | Postfix Notation<br>(Reverse Polish Notation) | 演算子がオペランドの後に位置する<br>計算機での処理が容易 |

| 用語             | 日本語         | 英语                  | 中国語   | 説明                                                                       |
|:-----------------|:---------------|:----------------------|:---------|:---------------------------------------------------------------------------|
| AI               | 人工知能       | Artificial Intelligence | 人工智能 | 人間の知的なふるまいを、コンピュータに再現する技術                           |
| 機械学習         | 機械学習       | Machine Learning      | 机器学习 | 人間が持つ学習能力を、コンピュータに再現する技術。AI技術の1つ                 |
| ニューラルネットワーク | 神経ネットワーク | Neural Network        | 神经网络 | 人間の脳の中にある神経回路が信号をやり取りする仕組みをコンピュータに再現する技術 |
| ディープラーニング   | 深層学習       | Deep Learning         | 深度学习 | ニューラルネットワークを何重にも重ねた構造を持つ機械学習                     |